cmake/modules/CheckCompiler.cmake

```
@@ -201,6 +201,19 @@ endif()
 
 
 #---Setup details depending on the major platform type----------------------------------------------
+
+function(ROOT_SET_COMPILER_FLAGS VARNAME FLAGS DOCSTRING)
+    # Don't set ROOT's defaults if it's overwritten, e.g. via setting CMAKE_CXX_FLAGS_DEBUG
+    # The *_INIT versions may have additional whitespaces.
+    STRING(STRIP ${${VARNAME}} COMPARE_VAR)
+    STRING(STRIP ${${VARNAME}_INIT} COMPARE_VAR_INIT)
+    if (COMPARE_VAR STREQUAL COMPARE_VAR_INIT)
+        set(${VARNAME} "${FLAGS}" CACHE STRING "${DOCSTRING}" FORCE)
+    else()
+        message(STATUS "Compiler flags for ${VARNAME} are not set to ROOT's default values because the variable was explicitely set.")
```

I would disable this message on standard message level, because when users set them, they want to set them.
```suggestion
        message(DEBUG "Compiler flags for ${VARNAME} are not set to ROOT's default values because the variable was explicitely set.")
```

---

Sound good :+1: